libcaf_net/caf/net/web_socket/framing.cpp

```diff
@@ -95,26 +112,58 @@ ptrdiff_t framing::consume(byte_span buffer, byte_span) {
       return -1;
     }
     // Reject any payload that exceeds max_frame_size. This covers assembled
-    //  payloads as well by including payload_buf_.
+    // payloads as well by including payload_buf_.
     if (payload_buf_.size() + hdr.payload_len > max_frame_size) {
       CAF_LOG_DEBUG("fragmented WebSocket payload exceeds maximum size");
       abort_and_shutdown(sec::protocol_error, "fragmented WebSocket payload "
                                               "exceeds maximum size");
       return -1;
     }
   }
-  // Wait for more data if necessary.
+  // Calculate at what point of the received buffer the delta payload begins.
+  auto offset = static_cast<ptrdiff_t>((buffer.size() - delta.size()))
+                - hdr_bytes;
+  // Offset < zero  - the delta buffer contains header bytes.
+  // Delta is empty - we didn't process the whole input last time we got called.
+  if (delta.empty() || offset < 0)
+    offset = 0;
```

Comment:
`delta` should never be empty. This would mean that the transport is calling you twice with the same data. Is this check only due to #1537?

Reply:
#1537 is one cause of this, as you stated, but the other valid scenario is if multiple frames arrive at once from the transport layer, and we process one frame per call. We have a unit test for this [here](https://github.com/actor-framework/actor-framework/blob/5933a5f7612ab93d45ede32f91c2e1a1f76f360c/libcaf_net/tests/legacy/net/web_socket/framing.cpp#L234)

Reply:
This is fixed now.